### PR TITLE
chore: update wallet and transfer docs

### PR DESCRIPTION
**Pull request by oetyng**
Also an attempt at better naming for wallet variable of created dbcs.
Still not entirely satisfactory though..

### CODE REVIEW

File: safenode/src/protocol/types/transfers/mod.rs [hunk 1]
```diff
@@ -131,6 +155,8 @@ fn transfer(send_inputs: Inputs) -> Result<Outputs> {
 
     let mut change_dbc = None;
     created_dbcs.retain(|created| {
+        // This is not 100% solid in my mind, since we could have another output that is sending to ourselves.
+        // In the wallet (which calls this code) such an operation is skipped. But worth noting anyway.
         if created.dbc.public_address() == &change_to && change.as_nano() > 0 {
             change_dbc = Some(created.dbc.clone());
             false
```
Comment (grumbach) [inline safenode/src/protocol/types/transfers/mod.rs:162]: If any other DBCs are for us, we will remove them
```
             change_dbc = Some(created.dbc.clone());
             false
```
An overwrite the `change_dbc` variable. 
Effectively removing the DBC from both `created_dbcs` and `change_dbc`.

File: safenode/src/protocol/types/wallet/local_store.rs [hunk 2]
```diff
@@ -146,10 +149,14 @@ impl<C: SendClient + Send + Sync + Clone> SendWallet<C> for LocalSendWallet<C> {
...
-            available_dbcs.push((dbc.clone(), derived_key));
+            if let Ok(derived_key) = dbc.derived_key(&self.wallet.key) {
+                available_dbcs.push((dbc.clone(), derived_key));
+            } else {
+                warn!(
+                    "Skipping DBC {:?} because we don't have the key to spend it",
+                    dbc.id()
+                );
+            }
         }
```
Comment (grumbach) [inline safenode/src/protocol/types/wallet/local_store.rs:160]: Having DBCs that are not ours in our wallet should be a bug right? While having `expect` makes our code panic prone and we should indeed remove them, we should also note that this change potentially hides this bug from our eyes. 
Or is this a feature?

**Review by grumbach — CHANGES_REQUESTED** [review]
The `change_dbc` bug is important IMO we should fix it.
One quick and dirty way to do it is to take the first DBC in our name and leave all the others in `created_dbcs` so we at least don't make DBCs disappear.

File: safenode/src/protocol/types/wallet/local_store.rs (hunk 2 above)
Comment (oetyng) [reply safenode/src/protocol/types/wallet/local_store.rs:160]: The wallet we have implemented does not add dbcs that are not for us.
We don't use expect and panics in production code, not for bugs either.

So, this is unreachable code for us. And it just needs some handling to compile.. so a warn log..

**Comment by oetyng** [comment]
> The `change_dbc` bug is important IMO we should fix it. One quick and dirty way to do it is to take the first DBC in our name and leave all the others in `created_dbcs` so we at least don't make DBCs disappear.

Yeah this is old code. It's not possible to get there since all layers above prevent it. So it's not as issue.
But, yeah as I noted, it is not solid.

**Pull request merged**